Add emission to transmodl api (#6628)
opentripplanner/OpenTripPlanner: doc/templates/sandbox/Emission.md

# CO₂ Emissions calculation

## Contact Info

- Digitransit Team
- Entur AS

## Documentation

> The emission module enable import of CO₂ Emissions data witch OTP can use to decorate itineraries

Reviewer: ```suggestion
The emission module enables import of CO₂ Emissions data which OTP can use to decorate itineraries
```

> and itinerary legs. Emissions data is loaded during the graph build process. OTP support two data

Reviewer: ```suggestion
and itinerary legs. Emissions data is loaded during the graph build process. OTP supports two data
```

import formats (yet to be standarized).

- The _Route format_ provided by HSL decorate each route with a avegare co2 emission per kilometers
  per person. The input parameters are `route_id`, `avg_co2_per_vehicle_per_km` and
  `avg_passenger_count`.

> - The _Trip format_ provided by Entur decorate each trip hop. The plan is to expand this to include

Reviewer: ```suggestion
- The _Trip format_ provided by Entur decorates each trip hop. The plan is to expand this to include
```

>   service date. This allow the emission provider to calculate acurate data down to each trip

Reviewer: ```suggestion
  service date. This allows the emission provider to calculate accurate data down to each trip
```

>   depature (except depatures operated by more than one vhiechle). The input parameters are 

Reviewer: ```suggestion
  depature (except depatures operated by more than one vehicle). The input parameters are 
```

>   depature (except depatures operated by more than one vehicle). The input parameters are 

Reviewer: ```suggestion
  departure (except departures operated by more than one vehicle). The input parameters are 
```

For example:

```csv
route_id,avg_co2_per_vehicle_per_km,avg_passenger_count
1234,123,20
2345,0,0
3456,12.3,20.0
```

### Emission trip hop format input data files
(by the Entur team)

The emissions are represented in grams per kilometer (g/Km) unit.

Emission file has the following columns:

| CSV Header                 | Description                                                                                                                                                                                                                                |
|:---------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `trip_id`                  | The GTFS trip id or the NeTEx ServiceJourney id                                                                                                                                                                                            |

> | `from_stop_id`             | The GTFS stop id or the NeTEx Quey id. This is use together with the `from_stop_sequence` to match the stop. If the emissions data do not match the given transit feed, then the row is dropped and an issue is added to the build report. |

Reviewer 2: ```suggestion
| `from_stop_id`             | The GTFS stop id or the NeTEx Quay id. This is use together with the `from_stop_sequence` to match the stop. If the emissions data does not match the given transit feed, then the row is dropped and an issue is added to the build report. |
```

> | `from_stop_sequence`       | The boardins stop sequence number in the trip stop pattern. The first stop is number one(1), not zero(0).                                                                                                                                  |

Reviewer 2: ```suggestion
| `from_stop_sequence`       | The boarding stop sequence number in the trip stop pattern. The first stop is number one(1), not zero(0).                                                                                                                                  |
```

| `co2`                      | Average carbon dioxide equivalent value for the vehicles used on given trip hop starting at the given stop. The unit is grams per person per hop.                                                                                          |

For example:

```csv
trip_id,from_stop_id,from_stop_sequence,co2
Trip:3,Stop:A,1,90.456
Trip:3,Stop:A,2,130.483
Trip:3,Stop:A,3,172.646
```

### Configuration

> To enable this functionality, you need to enable the "Co2Emissions" feature in the

Reviewer 2: ```suggestion
To enable this functionality, you need to enable the "Emission" feature in the
```

`otp-config.json` file.

```JSON
//otp-config.json
{

>   "Co2Emissions": true

Reviewer 2: ```suggestion
  "Emission": true
```
